quotes.py

```diff
@@ -0,0 +1,233 @@
+            UserID = reply.from_id
+        else:
+            if reply.fwd_from.from_name:
+                ProfilePhotoURL = ""
+                Username = reply.fwd_from.from_name
+            elif reply.fwd_from.channel_id:
+                try:
+                    cid = reply.fwd_from.channel_id
+                    resolved = (await self.client(telethon.tl.functions.channels.GetFullChannelRequest(cid))).chats[0]
+
+                    if isinstance(resolved.photo, telethon.tl.types.ChatPhotoEmpty):  # get_profile_photos won't work
+                        ProfilePhotoURL = ""
+
+                    AdminTitle = "Channel"
+                    Username = telethon.utils.get_display_name(resolved)
+                    UserID = cid
+                    Channel = resolved
+                except telethon.errors.rpcerrorlist.ChannelPrivateError:
+                    # we don't know anything about this channel.
+                    ProfilePhotoURL = ""
+                    Username = "Hidden."
+                AdminTitle = "Channel"  # 100%
+            elif reply.fwd_from.from_id:
+                try:
+                    UserID = reply.fwd_from.from_id
+                    resolved = (await self.client(telethon.tl.functions.users.GetFullUserRequest(UserID))).user
+
+                    if not await self.client.get_profile_photos(UserID):
+                        ProfilePhotoURL = ""
+
+                    Username = telethon.utils.get_display_name(resolved)
+                    AdminTitle = ""
+                except ValueError:
+                    ProfilePhotoURL = ""
+                    Username = "A user."
+                    AdminTitle = "User"
+
+        if ProfilePhotoURL != "" or Channel:
+            with BytesIO() as pfp:
+                if Channel:
+                    await self.client.download_profile_photo(Channel, pfp)
+                else:
+                    await self.client.download_profile_photo(UserID, pfp)
+                pfp.seek(0)
+                ProfilePhotoURL = "data:image/png;base64, " + base64.b64encode(pfp.read()).decode()
+                pfp.close()
+
+        if UserID is not None:
+            UsernameColor = self.config["USERNAME_COLORS"][UserID % 7]
+        else:
+            UsernameColor = self.config["DEFAULT_USERNAME_COLOR"]
+
+        request = json.dumps({
+            "ProfilePhotoURL": ProfilePhotoURL,
+            "UsernameColor": UsernameColor,
+            "Username": Username,
+            "AdminTitle": AdminTitle,
+            "Text": reply.message,
+            "Markdown": get_markdown(reply),
+            "Template": args[0],
+            "APIKey": self.config["API_TOKEN"]
+        })
+
+        resp = await utils.run_sync(requests.post, self.config["API_URL"] + "/api/v2/quote", data=request)
+        resp = await utils.run_sync(resp.json)
+
+        if resp["status"] == 500:  # API RETURNS 200 STATUS CODE ALL TIME
```

Comment:
Does the comment have to be quite so shouty?

Reply:
This if needs an `elif resp["status"] != 200`

Reply:
nooooooooo then it won't handle other errors